File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public static int[] solution(int n) {
        List<Integer> answer = new ArrayList<>();
        answer.add(n);
        
        while(true) {
            if (n == 1) break;

            if (n%2 == 0) {
                n = n/2;
                answer.add(n);
            } else {
                n = n*3+1;
                answer.add(n);
            }
        }

        return answer.stream().mapToInt(e -> e).toArray();
    }
}

File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n):
    answer = []
    answer.append(n)
    
    while(1):
        if n==1: break;
        if n%2: 
            n = n*3+1
            answer.append(n)
        else:
            n = n/2
            answer.append(n)
    
    return answer